lib/src/DOtherSide.cpp

```diff
@@ -287,12 +288,12 @@ void dos_qguiapplication_download_image(const char *imageSource, const char *fil
 
 void dos_qguiapplication_download_imageByUrl(const char *url, const char *filePath)
 {
-    auto manager = new QNetworkAccessManager;
-    manager->setAutoDeleteReplies(true);
-    QObject::connect(manager, &QNetworkAccessManager::finished, [manager, filePath](QNetworkReply *reply) {
+    static thread_local QNetworkAccessManager manager;
+    manager.setAutoDeleteReplies(true);
+    QObject::connect(&manager, &QNetworkAccessManager::finished, [path = QString(filePath)](QNetworkReply *reply) mutable {
```

Comment:
If we're reusing the QNetworkAccessManager would be better to connect this function to the reply object received in manager.get(QNetworkRequest(QUrl(url)));.
The QNetworkAccessManager::finished will be emitted on each request finished and previous connections are not removed.

Reply:
We could simply do: `reply->deleteLater();` at the end of the lambda

Reply:
(in both functions, instead of `manager.setAutoDeleteReplies(true);`)

Reply:
Thanks @alexjba, good catch! It should be connected to response indeed. When it comes to deleting replies, `manager.setAutoDeleteReplies(true);` seems cleaner for me than `reply->deleteLater();`, I would leave it as it is.

Reply:
OK, fair enough

Reply:
Ah, I see, it will be set again every time...

Reply:
I didn't get your point, sorry.

Reply:
So in that case we can use request attribute for that: `QNetworkRequest::AutoDeleteReplyOnFinishAttribute`

Reply:
That's what `manager.setAutoDeleteReplies(true);` does right?